### PR TITLE
[Misc] Merge test utils.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

Rename tests/pytests_utils/ -> tests/utils/, and merge tests/python/common/utils.py to tests/pytests_utils/checks.py.
In the future global utils for pytest should be in this dir.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by mufeili** [comment]
@frozenbugs Could you add a summary of what you did in this PR and its connection to #5438 ? This will also be helpful if future people need to understand what happened and the motivations behind.

File: tests/utils/checks.py [hunk 1]
```diff
@@ -66,3 +73,52 @@ def assert_is_identical_hetero(g, g2, ignore_internal_data=False):
         assert len(g.edges[etype].data) == len(g2.edges[etype].data)
         for k in g.edges[etype].data:
             assert F.allclose(g.edges[etype].data[k], g2.edges[etype].data[k])
+
+
+def check_graph_equal(g1, g2, *, check_idtype=True, check_feature=True):
+    assert g1.device == g1.device
```
Comment (mufeili) [inline tests/utils/checks.py:79]:
```suggestion
    assert g1.device == g2.device
```

File: tests/utils/checks.py [hunk 2]
```diff
@@ -66,3 +73,52 @@ def assert_is_identical_hetero(g, g2, ignore_internal_data=False):
...
+    assert g1.dsttypes == g2.dsttypes
+    assert g1.canonical_etypes == g2.canonical_etypes
+    assert g1.batch_size == g2.batch_size
+
+    # check if two metagraphs are identical
+    for edges, features in g1.metagraph().edges(keys=True).items():
+        assert g2.metagraph().edges(keys=True)[edges] == features
+
+    for nty in g1.ntypes:
+        assert g1.number_of_nodes(nty) == g2.number_of_nodes(nty)
```
Comment (mufeili) [inline tests/utils/checks.py:94]:
```suggestion
        assert g1.num_nodes(nty) == g2.num_nodes(nty)
```

File: tests/utils/checks.py [hunk 3]
```diff
@@ -66,3 +73,52 @@ def assert_is_identical_hetero(g, g2, ignore_internal_data=False):
...
+        assert F.allclose(g1.batch_num_nodes(nty), g2.batch_num_nodes(nty))
+    for ety in g1.canonical_etypes:
+        assert g1.number_of_edges(ety) == g2.number_of_edges(ety)
```
Comment (mufeili) [inline tests/utils/checks.py:97]:
```suggestion
        assert g1.num_edges(ety) == g2.num_edges(ety)
```

File: tests/utils/checks.py [hunk 4]
```diff
@@ -66,3 +73,52 @@ def assert_is_identical_hetero(g, g2, ignore_internal_data=False):
...
+            assert F.allclose(dst1, dst2)
+            assert F.allclose(eid1, eid2)
+        else:
+            assert F.allclose(src1, F.astype(src2, g1.idtype))
+            assert F.allclose(dst1, F.astype(dst2, g1.idtype))
+            assert F.allclose(eid1, F.astype(eid2, g1.idtype))
+
+    if check_feature:
+        for nty in g1.ntypes:
+            if g1.number_of_nodes(nty) == 0:
```
Comment (mufeili) [inline tests/utils/checks.py:112]:
```suggestion
            if g1.num_nodes(nty) == 0:
```

File: tests/utils/checks.py [hunk 5]
```diff
@@ -66,3 +73,52 @@ def assert_is_identical_hetero(g, g2, ignore_internal_data=False):
...
+                continue
+            for feat_name in g1.nodes[nty].data.keys():
+                assert F.allclose(
+                    g1.nodes[nty].data[feat_name], g2.nodes[nty].data[feat_name]
+                )
+        for ety in g1.canonical_etypes:
+            if g1.number_of_edges(ety) == 0:
```
Comment (mufeili) [inline tests/utils/checks.py:119]:
```suggestion
            if g1.num_edges(ety) == 0:
```

**Review by mufeili — COMMENTED** [review]
done a pass, some minor comments

File: tests/utils/checks.py (hunk 2 above)
Comment (frozenbugs) [reply tests/utils/checks.py:94]: Not related to this PR, will have a separate PR to fix them all.

File: tests/utils/checks.py (hunk 3 above)
Comment (frozenbugs) [reply tests/utils/checks.py:97]: ditto

File: tests/utils/checks.py (hunk 4 above)
Comment (frozenbugs) [reply tests/utils/checks.py:112]: ditto

File: tests/utils/checks.py (hunk 5 above)
Comment (frozenbugs) [reply tests/utils/checks.py:119]: ditto

File: tests/utils/checks.py (hunk 1 above)
Comment (frozenbugs) [reply tests/utils/checks.py]: Good catch, 3 years old bug :)

**Comment by frozenbugs** [comment]
> @frozenbugs Could you add a summary of what you did in this PR and its connection to #5438 ? This will also be helpful if future people need to understand what happened and the motivations behind.

Done.

**Review by mufeili — APPROVED** [review]

**Pull request merged**